modules/commerce_bluesnap_currency/src/Form/CommerceBluesnapCurrencyResolverConversion.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Drupal\commerce_bluesnap_currency\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\commerce_currency_resolver\Form\CommerceCurrencyResolverConversion;
+
+/**
+ * Class CommerceCurrencyResolverConversion.
+ *
+ * @package Drupal\commerce_currency_resolver\Form
+ */
+class CommerceBluesnapCurrencyResolverConversion extends CommerceCurrencyResolverConversion {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    // Get current settings.
+    $config = $this->config('commerce_currency_resolver.currency_conversion');
+
+    // Call the parent form.
+    $form = parent::buildForm($form, $form_state);
+
+    // Unset the form submit coming from parent form as
+    // its been added below.
+    unset($form['submit']);
+
+    // Bluesnap exchange rate config.
+    $form['bluesnap'] = [
+      '#type' => 'details',
+      '#title' => t('Bluesnap Settings'),
+      '#open' => FALSE,
+      '#tree' => TRUE,
+      '#states' => [
+        'visible' => [
+          ':input[name="source"]' => ['value' => 'exchange_rate_bluesnap'],
+        ],
+      ],
+    ];
+    $form['bluesnap']['username'] = [
+      '#type' => 'textfield',
+      '#title' => t('Username'),
+      '#default_value' => $config->get('bluesnap')['username'],
+    ];
+    $form['bluesnap']['password'] = [
+      '#type' => 'textfield',
+      '#title' => t('Password'),
+      '#default_value' => $config->get('bluesnap')['password'],
+    ];
```

Comment:
**CHANGE REQUEST**

Better use the `password` form element instead of textfield.

Reply:
Used password field